python/cuml/common/array.py

```diff
@@ -223,18 +226,29 @@ def to_output(self, output_type='cupy'):
                 output_type = 'dataframe'
 
         if output_type == 'cupy':
+            if output_dtype and self.dtype != output_dtype:
+                return cp.asarray(self, dtype=output_dtype)
             return cp.asarray(self)
 
         elif output_type == 'numba':
+            if output_dtype and self.dtype != output_dtype:
+                return cuda.as_cuda_array(cp.asarray(self, dtype=output_dtype))
             return cuda.as_cuda_array(self)
 
         elif output_type == 'numpy':
+            if output_dtype and self.dtype != output_dtype:
+                return cp.asnumpy(
+                    cp.asarray(self, dtype=output_dtype), order=self.order
+                )
             return cp.asnumpy(cp.asarray(self), order=self.order)
 
         elif output_type == 'dataframe':
             if self.dtype not in [np.uint8, np.uint16, np.uint32,
                                   np.uint64, np.float16]:
-                mat = cuda.as_cuda_array(self)
+                if output_dtype and self.dtype != output_dtype:
```

Comment:
```suggestion
                if output_dtype:
```

Second check shouldn't be needed if I'm not mistaken

Reply:
Should be resolved by https://github.com/rapidsai/cuml/pull/2431/commits/4f46c731c03ced34a8bf6210d58c72c6166ca2e7